packages/mattermost-redux/src/selectors/entities/search.ts

```diff
@@ -26,3 +27,28 @@ export const getAllUserMentionKeys: (state: GlobalState) => UserMentionKey[] = c
         return userMentionKeys.concat(groupMentionKeys);
     },
 );
+
+export function recentSearches(state: GlobalState): SearchParams[] {
+    const recentSearches = state.entities.search.recentSearches;
+    const channels = state.entities.channels.channels;
+    const userProfiles = state.entities.users.profiles;
+    if (!recentSearches) {
+        return [];
+    }
+    return recentSearches.map((searchParams) =>
```

Comment:
I think this can be turned into a selector so we can store the recentSearches in the state and help reduce some complexity in the component as well as avoid rerenders.

Reply:
I've replaced this with a `createSelector` as explained by @hmhealey :+1: